swarm/storage/ldbstore.go

```diff
@@ -581,13 +578,13 @@ mainLoop:
 					close(done)
 				}()
 
-				e = s.entryCnt
 				select {
 				case <-s.quit:
 					s.lock.Unlock()
 					break mainLoop
 				case <-done:
 				}
+				e = s.entryCnt
```

Comment:
`e` is meaningful only in the context of this for loop, so we care about updating it, only after `s.collectGarbage()` is done.